### ISSUE TITLE
modify find_free_prefix to be able to specify a starting point for finding a 

### ISSUE THREAD

**Issue by dkelson** [SpriteLink/NIPAP#1227]
Let's say I have a prefix 100.64.0.0/20 I would like to be request an single IP for a host on the next available IP above 10.64.1.0.

**Comment by garberg** [comment]
You can pass any prefix to find_free_prefix, it does not need to be defined in NIPAP before. So in your case, you could simply pass 10.64.1.0/24 or similar to find_free_prefix.

Would that solve your problem?

**Comment by dkelson** [comment]
Can you do this with the nipap cli?

Suppose 100.64.0.0/20 is part of a pool called "backbone-ips"

On the network it is actually a /20. But we want the bottom half to be for infrastructure and and the top half for servers.

Would this work? 

nipap address add family ipv4 from-pool backbone-ips description "server 003" type host prefix 100.64.8.0/21

or how about if the pool also contains an IPv6 netblock, would this work?

nipap address add family dual-stack from-pool backbone-ips description "server 003" add-hosts server03 prefix 100.64.8.0/21

**Comment by plajjan** [comment]
@dkelson no, that wouldn't work. Networks that exist in the network, like that /20 which is configured on interfaces, are supposed to be of type "assignment" in NIPAP and you can't currently subdivide an assignment. The point is to define these things in the system so that you don't have to "think so much" when you request something from a pool.

This isn't the first time this request has popped up, well, not in exactly this form, but it's been the same underlying problem - which is to create divisions within the definition in NIPAP of an IP network ("assignment"). I think it should be addressed by allowing some division construct in NIPAP rather than forcing it upon the user to always include extra information in the request.
